### PR TITLE
newsletter subscribe with mailchimp integration completed

### PR DESCRIPTION
**Pull request by bhavdip111**

### CODE REVIEW

File: app/api/users/news-letter/route.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    const response = await mailchimp.lists.addListMember(listId, {
+      email_address: data.email,
+      status: 'subscribed',
+    })
+
+    if (response.status !== 'subscribed') {
+      return new Response(null, { status: 400 })
+    }
+
+    return new Response(null, { status: 200 })
```
Comment (krunalinfynno) [inline app/api/users/news-letter/route.tsx]: Use NextResponse

File: app/api/users/news-letter/route.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  } catch (error: any) {
+    console.log('error', error)
+    if (error instanceof z.ZodError) {
+      return new Response(JSON.stringify(error.issues), { status: 422 })
+    }
+
+    return new Response(null, { status: 500 })
```
Comment (krunalinfynno) [inline app/api/users/news-letter/route.tsx]: Use Next Response

File: app/api/users/news-letter/route.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    if (response.status !== 'subscribed') {
+      return new Response(null, { status: 400 })
+    }
+
+    return new Response(null, { status: 200 })
+  } catch (error: any) {
+    console.log('error', error)
+    if (error instanceof z.ZodError) {
+      return new Response(JSON.stringify(error.issues), { status: 422 })
```
Comment (krunalinfynno) [inline app/api/users/news-letter/route.tsx]: Use Next Response

File: components/news-letter/news-letter-form.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    resolver: zodResolver(newsLetterSchema),
+    defaultValues: {
+      email: '',
+    },
+  })
+
+  const handleSubmit = (values: z.infer<typeof newsLetterSchema>) => {
+    mutate(values, {
+      onSuccess: () => {
+        toast({ title: 'You have subscribed to the news letter' })
```
Comment (krunalinfynno) [inline components/news-letter/news-letter-form.tsx]: news letter should be Newsletter

File: components/news-letter/news-letter.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            latest blogs and information about our launch first.
+          </p>
+          <NewsLetterForm />
+        </div>
+      </div>
+    </div>
+  )
+}
+
+export default NewsLetter
```
Comment (krunalinfynno) [inline components/news-letter/news-letter.tsx]: Move this to page.tsx no need for extra component.

- Newsletter not news letter

**Review by krunalinfynno — CHANGES_REQUESTED** [review]

File: components/news-letter/news-letter.tsx (hunk 5 above)
Comment (bhavdip111) [reply components/news-letter/news-letter.tsx]: This component is called in homepage

**Pull request merged**